src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1233,7 +1233,9 @@ internal static Activity Create(ActivitySource source, string name, ActivityKind
                     activity._parentSpanId = parentContext.SpanId.ToString();
                 }
 
-                activity.ActivityTraceFlags = parentContext.TraceFlags;
+                // Note: Don't inherit Recorded from parent as it is set below
+                // based on sampling decision
+                activity.ActivityTraceFlags = parentContext.TraceFlags & ~ActivityTraceFlags.Recorded;
```

Comment:
Why does this line exist at all? If W3C trace context introduces new flags, what is to say that inheriting the flags from the parent will be the correct thing to do for those new flags?

Reply:
Ah, I see this was already discussed... removing the line seems more correct to me...

Reply:
In looking at the latest editor's draft of the spec, there is a new flag proposed: [Random Trace ID Flag](https://w3c.github.io/trace-context/#random-trace-id-flag). My read of this flag is that it actually _should_ be inherited from the parent, but there's no way to be certain this will be true for all new flags.

Reply:
Keeping this line is not hurting for now. Any newly introduced flag will need to be decided if it is inheritable from the parent or not anyway.